test/e2e/helpers.go

```diff
@@ -21,16 +21,183 @@ package e2e
+	Eventually(func() bool {
+		deployment := &appsv1.Deployment{}
+		key := client.ObjectKey{Namespace: namespace, Name: name}
+		if err := k8sClient.Get(ctx, key, deployment); err != nil {
+			return false
+		}
+
+		for _, c := range deployment.Status.Conditions {
+			if c.Type == appsv1.DeploymentAvailable && c.Status == corev1.ConditionTrue {
+				return true
+			}
+		}
+
+		return false
+
+	}, timeout).Should(Equal(true))
+}
+
+func waitForDeploymentDeleted(namespace, name string, k8sClient client.Client) {
+	Eventually(func() bool {
+		deployment := &appsv1.Deployment{}
+		key := client.ObjectKey{Namespace: namespace, Name: name}
+		if err := k8sClient.Get(ctx, key, deployment); err != nil {
+			return apierrors.IsNotFound(err)
+		}
+
+		return false
+	}, timeout).Should(Equal(true))
+}
+
+func waitForInstalledVersionSet(provider genericprovider.GenericProvider, version string, k8sClient client.Client) {
+	Eventually(func() bool {
+		key := client.ObjectKey{Namespace: operatorNamespace, Name: provider.GetName()}
+		if err := k8sClient.Get(ctx, key, provider.GetObject()); err != nil {
+			return false
+		}
+
+		if provider.GetStatus().InstalledVersion != nil && *provider.GetStatus().InstalledVersion == version {
+			return true
+		}
+		return false
+	}, timeout).Should(Equal(true))
+}
+
+func cleanupAndWait(ctx context.Context, cl client.Client, objs ...client.Object) error {
+	if err := cleanup(ctx, cl, objs...); err != nil {
+		return err
+	}
+	// Makes sure the cache is updated with the deleted object
+	errs := []error{}
+	for _, o := range objs {
+		oCopy := o.DeepCopyObject().(client.Object)
+		key := client.ObjectKeyFromObject(o)
+		err := wait.ExponentialBackoff(
+			wait.Backoff{
+				Duration: 150 * time.Millisecond,
+				Factor:   1.5,
+				Steps:    8,
+				Jitter:   0.4,
+			},
+			func() (done bool, err error) {
+				if err := cl.Get(ctx, key, oCopy); err != nil {
+					if apierrors.IsNotFound(err) {
+						return true, nil
+					}
+					return false, err
+				}
+				return false, nil
+			})
+		errs = append(errs, errors.Wrapf(err, "key %s, %s is not being deleted from the client cache", o.GetObjectKind().GroupVersionKind().String(), key))
+	}
+	return kerrors.NewAggregate(errs)
+}
+
+func cleanup(ctx context.Context, cl client.Client, objs ...client.Object) error {
+	errs := []error{}
+	for _, o := range objs {
+		oCopy := o.DeepCopyObject().(client.Object)
+		key := client.ObjectKeyFromObject(oCopy)
+
+		err := cl.Get(ctx, key, oCopy)
+		if apierrors.IsNotFound(err) {
+			continue
+		}
+		errs = append(errs, err)
+
+		// Remove finalizers from the object
+		if oCopy.GetFinalizers() != nil {
+			oCopy.SetFinalizers(nil)
+		}
+
+		err = cl.Update(ctx, oCopy)
+		if apierrors.IsNotFound(err) {
+			continue
+		}
+		errs = append(errs, err)
+
+		err = cl.Delete(ctx, oCopy)
+		if apierrors.IsNotFound(err) {
+			continue
+		}
+		errs = append(errs, err)
+	}
+	return kerrors.NewAggregate(errs)
```

Comment:
Does this exclude nil errors? Looks like you might be adding some nil errors to this list possibly